assets/js/utils/style-attributes-utils.ts

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable @wordpress/no-unsafe-wp-apis */
+/**
+ * External dependencies
+ */
+import { __experimentalUseColorProps } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { isFeaturePluginBuild } from '../settings/blocks/feature-flags';
+import { isString, isObject } from '../types/type-guards';
+
+type WithClass = {
+	className: string;
+};
+
+type WithStyle = {
+	style: Record< string, unknown >;
+};
+
+const parseStyle = ( style: unknown ): Record< string, unknown > => {
+	if ( isString( style ) ) {
+		return JSON.parse( style ) || {};
+	}
+
+	if ( isObject( style ) ) {
+		return style;
+	}
+
+	return {};
+};
+
+const parseSpacingStyle = (
+	spacing: Record< string, unknown >
+): Record< string, unknown > => {
+	const keys = [ 'margin' ];
+
+	const getValueOrDefault = ( value: unknown ) => {
+		return isString( value ) && value.length > 0 ? value : '0';
+	};
+
+	return Object.keys( spacing ).reduce( ( acc, key ) => {
+		const spacingProperty = isObject( spacing[ key ] )
+			? ( spacing[ key ] as Record< string, unknown > )
+			: {};
+
+		if ( keys.includes( key ) ) {
+			return {
+				...acc,
+				[ key ]: `${ getValueOrDefault(
+					spacingProperty.top
+				) } ${ getValueOrDefault(
+					spacingProperty.right
+				) } ${ getValueOrDefault(
+					spacingProperty.bottom
+				) } ${ getValueOrDefault( spacingProperty.left ) }`,
+			};
```

Comment:
Same as above, we should only use shorthand when all four sides are set, otherwise, we should use single side rules.

Reply:
Thanks for this! I updated the function!